tests/models/test_modeling_common.py

```diff
@@ -866,6 +867,19 @@ def test_model_parallelism(self):
 
                 self.assertTrue(torch.allclose(base_output[0], new_output[0], atol=1e-5))
 
+    def test_loading_from_subfolder(self):
```

Comment:
not sure if this test will catch the error in https://github.com/huggingface/diffusers/pull/8416 since we only have issue when it loads from the legacy model class name, 
e.g. for pixart  `model_class = PixArtTransformer2DModel`

I left a comment here https://github.com/huggingface/diffusers/pull/7647/files#r1631917736
I think we should keep the tests with legacy class name so we will be abe to catch these errors